### PR TITLE
Fixed yii\db\Schema::getSchemaNames() for MSSQL (fixes #15108)

### LINKED ISSUE #15108: Yii 2.0.13 Gii broken for MSSQL
### What steps will reproduce the problem?
Go to gii model generator. Enter name of table in MSSQL database.
Click preview. 

### What is the expected result?
Preview of model is generated.

### What do you get instead?
Error:
SQLSTATE[42S02]: [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]Invalid object name 'pg_namespace'.
The SQL being executed was: SELECT ns.nspname AS schema_name
FROM pg_namespace ns
WHERE ns.nspname != 'information_schema' AND ns.nspname NOT LIKE 'pg_%'
ORDER BY ns.nspname

### Additional info

This appears to be an issue with protected function findSchemaNames(): https://github.com/yiisoft/yii2/blob/d0bf3cfeb4813cefbaa2e3ac0b6cbf0cbfa931c0/framework/db/mssql/Schema.php#L122-L129

As mssql doesn't doesn't store schemas in pg_namespace. The query should probably be using sys.schemas or INFORMATION_SCHEMA.SCHEMATA

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.13
| PHP version      | 7.1.1
| Operating system |Windows Server 2012 R2

### PR DESCRIPTION
**Pull request by sergeymakinen**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15108

### CODE REVIEW

File: framework/db/pgsql/Schema.php [hunk 1]
```diff
@@ -142,12 +142,13 @@ protected function resolveTableName($name)
      */
     protected function findSchemaNames()
     {
-        $sql = <<<'SQL'
-SELECT ns.nspname AS schema_name
-FROM pg_namespace ns
-WHERE ns.nspname != 'information_schema' AND ns.nspname NOT LIKE 'pg_%'
-ORDER BY ns.nspname
+        static $sql = <<<'SQL'
```
Comment (samdark) [inline framework/db/pgsql/Schema.php]: Doesn't seem to be MSSQL-related.

Comment (sergeymakinen) [reply framework/db/pgsql/Schema.php]: Just a findSchemaNames normalization. This PR handles all of supported DBMSes.

**Comment by sergeymakinen** [comment]
To be clear: this PR adds absent `getSchemaNames` tests and tweaks an SQL code in `findSchemaNames`.
It’s tested on all currently supported DBMSes.
As a result, it fixes the mentioned issue. Maybe even more. I can strip all non MSSQL related code and create a new PR but in *this* situation I don’t think it’s crucial.

**Review by SilverFire — APPROVED** [review]

File: framework/db/pgsql/Schema.php (hunk 1 above)
Comment (samdark) [reply framework/db/pgsql/Schema.php]: Then the changelog should be adjusted.

Comment (samdark) [reply framework/db/pgsql/Schema.php]: No need to strip anything from the PR.

**Review by samdark — APPROVED** [review]

**Comment by sergeymakinen** [comment]
Hold on, don't merge it please. I have found a better way to handle predefined schemas in MSSQL. Will commit ASAP.

**Comment by sergeymakinen** [comment]
Everything should be fine and ready now.

File: framework/db/pgsql/Schema.php (hunk 1 above)
Comment (sergeymakinen) [reply framework/db/pgsql/Schema.php]: Adjusted. I don't know, maybe "Fixed `yii\db\Schema::getSchemaNames()` for MSSQL, updated SQL queries and added tests for all DBMSes" sounds even better.

**Comment by samdark** [comment]
Perfect! Thank you very much.

**Pull request merged**